src/v/cluster/controller_api.cc

```diff
@@ -53,6 +53,37 @@ controller_api::get_reconciliation_state(std::vector<model::ntp> ntps) {
     });
 }
 
+ss::future<result<bool>>
+controller_api::all_reconciliations_done(std::vector<model::ntp> ntps) {
+    bool reduced = true;
+    errc reduced_err{errc::success};
+
+    // For a huge topic with e.g. 100k partitions, this will be a huge loop:
+    // that means we need parallelism, but not so much that we totally
+    // saturate inter-core queues.
+    co_await ss::max_concurrent_for_each(
+      ntps.begin(),
+      ntps.end(),
+      8192,
+      [this, &reduced, &reduced_err](const model::ntp& ntp) -> ss::future<> {
+          auto status = co_await get_reconciliation_state(ntp);
+          if (
+            status.cluster_errc() != errc::success
+            && reduced_err == errc::success) {
+              reduced_err = status.cluster_errc();
```

Comment:
can we `break` in this case ?

Reply:
I thought about that, but it seems like there isn't a way to do it with max_concurrent_for_each (even if I through an exception, it still iterates through whole range) -- do you have a neat way of doing it?

Reply:
yeah, it seems that there is no way to break the loop

Reply:
afaik you need to roll your own using parallel for each in batches.

Reply:
I ended up doing exactly that while taking another pass on this.